### PR TITLE
Issue 4

### PR DESCRIPTION
**Pull request by benjamin-work**
* Implement a `PrintLog` class.
* Add tests for callbacks

### CODE REVIEW

**Comment by benjamin-work** [comment]
Addresses #4 
@ottonemo pls review

File: inferno/callbacks.py [hunk 1]
```diff
@@ -139,11 +144,85 @@ def on_batch_end(self, net, X, y, train):
...
+        headers = [key for key in self.keys if not key.endswith('_best')]
+        return tabulate(
+            formatted,
+            headers=headers,
+            tablefmt=self.tablefmt,
+            floatfmt=self.floatfmt,
+        )
+
+    def on_epoch_end(self, net, *args, **kwargs):
+        data = net.history[-1:, self.keys]
```
Comment (ottonemo) [inline inferno/callbacks.py]: What if the key does not exist in the history? This would raise a `KeyError` here.

Comment (benjamin-work) [inline inferno/callbacks.py]: What would be the best solution here? There is no earlier point in time where we could detect missing keys.

Comment (ottonemo) [inline inferno/callbacks.py]: True, but maybe we handle this case a bit more gracefully? I.e. writing a helpful error message explaining the issue instead. For example: "Key <x> was not found in the history which indicates that the event never occurred or you misspelled the key name."?

Comment (benjamin-work) [inline inferno/callbacks.py]: Is this better now?

**Pull request merged**